trieste/models/gpflux/sampler.py

```diff
@@ -0,0 +1,170 @@
+
+        super().__init__(model)
+
+        self.model = model.model_gpflux
+
+        if not isinstance(self.model, DeepGP):
+            raise ValueError(
+                f"GPflux model must be a gpflux.models.DeepGP, received {type:self.model}"
+            )
+
+    def get_trajectory(self) -> TrajectoryFunction:
+        """
+        Generate an approximate function draw (trajectory) by using the GPflux sampling
+        functionality. These trajectories are differentiable with respect to the input, so can be
+        used to e.g. find the minima of Thompson samples.
+
+        :return: A trajectory function representing an approximate trajectory from the deep Gaussian
+            process, taking an input of shape `[N, D]` and returning shape `[N, 1]`
+        """
+
+        return sample_dgp(self.model)
+
+
+class DeepGaussianProcessReparamSampler(ReparametrizationSampler[GPfluxPredictor]):
+    r"""
+    This sampler employs the *reparameterization trick* to approximate samples from a
+    :class:`DeepGaussianProcess`\ 's predictive distribution.
+    """
+
+    def __init__(self, sample_size: int, model: ProbabilisticModel):
+        """
+        :param sample_size: The number of samples for each batch of points. Must be positive.
+        :param model: The model to sample from. Must be a :class:`GPfluxPredictor`
+        :raise ValueError (or InvalidArgumentError): If ``sample_size`` is not positive, or if
+            model is not a :class:`GPfluxPredictor`.
+        """
+        if not isinstance(model, GPfluxPredictor):
+            raise ValueError(
+                f"Model must be a gpflux.interface.GPfluxPredictor, received {type(model)}"
+            )
+
+        super().__init__(sample_size, model)
+
+        self.model = model.model_gpflux
+
+        # Each element of _eps_list is essentially a lazy constant. It is declared and assigned an
+        # empty tensor here, and populated on the first call to sample
+        self._eps_list = [
+            tf.Variable(tf.ones([sample_size, 0], dtype=tf.float64), shape=[sample_size, None])
+            for _ in range(len(self.model.f_layers))
+        ]
+
+    def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorType:
+        """
+        Return approximate samples from the `model` specified at :meth:`__init__`. Multiple calls to
+        :meth:`sample`, for any given :class:`DeepGaussianProcessReparamSampler` and ``at``, will
+        produce the exact same samples. Calls to :meth:`sample` on *different*
+        :class:`DeepGaussianProcessReparamSampler` instances will produce different samples.
+
+        :param at: Where to sample the predictive distribution, with shape `[N, D]`, for points
+            of dimension `D`.
+        :param jitter: The size of the jitter to use when stabilizing the Cholesky
+            decomposition of the covariance matrix.
+        :return: The samples, of shape `[S, N, L]`, where `S` is the `sample_size` and `L` is
```

Comment:
so, `ReparametrizationSampler` says:
```
:return: Samples of shape `[sample_size, D]`.
```

gpflow's `IndependentReparametrizationSampler` docs say smth sligthly different:
```
:return: The samples, of shape `[..., S, 1, L]`, where `S` is the `sample_size` and `L` is
            the number of latent model dimensions.
```

I'm not sure what should be returned here actually... @henrymoss any thoughts?

Reply:
Yes, this is quite confusing. Would be good to have input from @henrymoss 

I think it should be [S, N, L], it doesn't really make sense in the `ReparametrizationSampler` to say `[S, D]` because then `N` disappears and `D` is the input dim.

Reply:
eeek. YEah, [S, N, L] looks correct.

Reply:
Thanks, should we make a separate PR to fix `IndependentReparametrizationSampler` then?

Reply:
its just fixing the docs slightly, it can go in this PR I would say